include/coro/thread_pool.hpp

```diff
@@ -119,11 +121,22 @@ class thread_pool
     [[nodiscard]] auto schedule() -> schedule_operation;
 
     /**
-     * Spawns the given task to be run on this thread pool, the task is detached from the user.
+     * Spawns the given task to be run on this thread pool, the task is detached from the user and cannot be joined.
      * @param task The task to spawn onto the thread pool.
      * @return True if the task has been spawned onto this thread pool.
      */
-    auto spawn(coro::task<void>&& task) noexcept -> bool;
+    auto spawn_detached(coro::task<void>&& task) noexcept -> bool;
+
+    /**
+     * Spawns the given task to be run on this thread pool, the task returned must be joined in the future.
+     * @note The returned task shouldn't be co_await'ed immediately, the spawned task is started on the thread
+     *       pool automatically but the returned join task *must* be co_await'ed at some point in the future.
+     *       If you drop the returned task it will hang the thread until the spawned task completes so it is
+     *       highly advisable to co_await the returned join task appropriately.
```

Comment:
Add that `spawn_joinable` is not itself a coroutine. And maybe also that `spawn_detached` is preferable because of a lower overhead, if you just need fire and forget.